cpu/stm32/periph/spi.c

```diff
@@ -64,6 +68,23 @@ static inline bool _use_dma(const spi_conf_t *conf)
 }
 #endif
 
+static uint8_t _get_clkdiv(const spi_conf_t *conf, uint32_t clock)
+{
+    uint32_t bus_clock = periph_apb_clk(conf->apbbus);
+    uint8_t div = bus_clock / (2 * clock);
```

Comment:
If you require a speed of 100KHz with 32MHz bus_clock then div = 320 which is truncated to 64 because of uint8_t so you will end up with 500KHz speed.

Also for 128 < div < 255 the function return 8 which is overflow the BR register on the SPE bit and result in bus_clock/2 speed (BR bits are effectively set to 0).

Reply:
Thanks for the feedback. I've reworked the function a bit taking your feedback into account. I've also added some `DEBUG` logging printing the resulting clock speed.